Compiling axum 0.2.4 with rust 1.55

## Bug Report

<!--
Thank you for reporting an issue.

Please fill in as much of the template below as you're able.
-->



### Version

axum 0.2.4

<!--
List the versions of all `axum` crates you are using. The easiest way to get
this information is using `cargo tree`:

`cargo tree | grep axum`
-->

### Platform

<!---
Output of `uname -a` (UNIX), or version and 32 or 64-bit (Windows)
-->
`Linux lenovo 5.11.0-34-generic #36-Ubuntu SMP Thu Aug 26 19:22:09 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

### Crates

<!--
If known, please specify the affected axum crates. Otherwise, delete this
section.
-->

### Description

axum cannot compile. i just added it to cargo file, not used anywhere. Just plain build

```
Compiling axum v0.2.4
error[E0309]: the parameter type `Res` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:301:64
    |
292 |   impl<F, Fut, Res, B> Handler<B, ()> for F
    |                --- help: consider adding an explicit lifetime bound...: `Res: 'async_trait`
...
301 |       async fn call(self, _req: Request<B>) -> Response<BoxBody> {
    |  ________________________________________________________________^
302 | |         self().await.into_response().map(box_body)
303 | |     }
    | |_____^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds

error[E0309]: the parameter type `Fut` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:301:64
    |
292 |   impl<F, Fut, Res, B> Handler<B, ()> for F
    |           --- help: consider adding an explicit lifetime bound...: `Fut: 'async_trait`
...
301 |       async fn call(self, _req: Request<B>) -> Response<BoxBody> {
    |  ________________________________________________________________^
302 | |         self().await.into_response().map(box_body)
303 | |     }
    | |_____^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds

error[E0309]: the parameter type `T16` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                                               |
    |   |                                                                               help: consider adding an explicit lifetime bound...: `T16: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T15` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                                          |
    |   |                                                                          help: consider adding an explicit lifetime bound...: `T15: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T14` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                                     |
    |   |                                                                     help: consider adding an explicit lifetime bound...: `T14: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T13` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                                |
    |   |                                                                help: consider adding an explicit lifetime bound...: `T13: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T12` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                           |
    |   |                                                           help: consider adding an explicit lifetime bound...: `T12: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T11` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                      |
    |   |                                                      help: consider adding an explicit lifetime bound...: `T11: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T10` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                 |
    |   |                                                 help: consider adding an explicit lifetime bound...: `T10: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T9` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                             |
    |   |                                             help: consider adding an explicit lifetime bound...: `T9: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T8` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                         |
    |   |                                         help: consider adding an explicit lifetime bound...: `T8: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T7` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                     |
    |   |                                     help: consider adding an explicit lifetime bound...: `T7: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T6` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                 |
    |   |                                 help: consider adding an explicit lifetime bound...: `T6: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T5` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                             |
    |   |                             help: consider adding an explicit lifetime bound...: `T5: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T4` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                         |
    |   |                         help: consider adding an explicit lifetime bound...: `T4: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T3` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                     |
    |   |                     help: consider adding an explicit lifetime bound...: `T3: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T2` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                 |
    |   |                 help: consider adding an explicit lifetime bound...: `T2: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T1` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |             |
    |   |             help: consider adding an explicit lifetime bound...: `T1: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `Res` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
313 |           impl<F, Fut, B, Res, $head, $($tail,)*> Handler<B, ($head, $($tail,)*)> for F
    |                           --- help: consider adding an explicit lifetime bound...: `Res: 'async_trait`
...
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   ------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `Fut` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
313 |           impl<F, Fut, B, Res, $head, $($tail,)*> Handler<B, ($head, $($tail,)*)> for F
    |                   --- help: consider adding an explicit lifetime bound...: `Fut: 'async_trait`
...
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   ------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0309`.
error: could not compile `axum` due to 170 previous errors
.../cargo-registry  master [✘!?] is 📦 v0.1.0 via 🦀 v1.55.0 $ cargo  b        
   Compiling axum v0.2.4
error[E0309]: the parameter type `Res` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:301:64
    |
292 |   impl<F, Fut, Res, B> Handler<B, ()> for F
    |                --- help: consider adding an explicit lifetime bound...: `Res: 'async_trait`
...
301 |       async fn call(self, _req: Request<B>) -> Response<BoxBody> {
    |  ________________________________________________________________^
302 | |         self().await.into_response().map(box_body)
303 | |     }
    | |_____^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds

error[E0309]: the parameter type `Fut` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:301:64
    |
292 |   impl<F, Fut, Res, B> Handler<B, ()> for F
    |           --- help: consider adding an explicit lifetime bound...: `Fut: 'async_trait`
...
301 |       async fn call(self, _req: Request<B>) -> Response<BoxBody> {
    |  ________________________________________________________________^
302 | |         self().await.into_response().map(box_body)
303 | |     }
    | |_____^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds

error[E0309]: the parameter type `T16` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                                               |
    |   |                                                                               help: consider adding an explicit lifetime bound...: `T16: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T15` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                                          |
    |   |                                                                          help: consider adding an explicit lifetime bound...: `T15: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T14` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                                     |
    |   |                                                                     help: consider adding an explicit lifetime bound...: `T14: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T13` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                                |
    |   |                                                                help: consider adding an explicit lifetime bound...: `T13: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T12` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                           |
    |   |                                                           help: consider adding an explicit lifetime bound...: `T12: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T11` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                      |
    |   |                                                      help: consider adding an explicit lifetime bound...: `T11: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T10` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                                 |
    |   |                                                 help: consider adding an explicit lifetime bound...: `T10: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T9` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                             |
    |   |                                             help: consider adding an explicit lifetime bound...: `T9: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T8` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                         |
    |   |                                         help: consider adding an explicit lifetime bound...: `T8: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T7` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                     |
    |   |                                     help: consider adding an explicit lifetime bound...: `T7: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T6` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                                 |
    |   |                                 help: consider adding an explicit lifetime bound...: `T6: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T5` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                             |
    |   |                             help: consider adding an explicit lifetime bound...: `T5: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T4` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                         |
    |   |                         help: consider adding an explicit lifetime bound...: `T4: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T3` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                     |
    |   |                     help: consider adding an explicit lifetime bound...: `T3: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T2` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |                 |
    |   |                 help: consider adding an explicit lifetime bound...: `T2: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `T1` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   -------------------------------------------------------------------------------------
    |   |             |
    |   |             help: consider adding an explicit lifetime bound...: `T1: 'async_trait`
    |   in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `Res` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
313 |           impl<F, Fut, B, Res, $head, $($tail,)*> Handler<B, ($head, $($tail,)*)> for F
    |                           --- help: consider adding an explicit lifetime bound...: `Res: 'async_trait`
...
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   ------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0309]: the parameter type `Fut` may not live long enough
   --> /home/bastian/.cargo/registry/src/github.com-1ecc6299db9ec823/axum-0.2.4/src/handler/mod.rs:324:71
    |
313 |           impl<F, Fut, B, Res, $head, $($tail,)*> Handler<B, ($head, $($tail,)*)> for F
    |                   --- help: consider adding an explicit lifetime bound...: `Fut: 'async_trait`
...
324 |               async fn call(self, req: Request<B>) -> Response<BoxBody> {
    |  _______________________________________________________________________^
325 | |                 let mut req = crate::extract::RequestParts::new(req);
326 | |
327 | |                 let $head = match $head::from_request(&mut req).await {
...   |
341 | |                 res.into_response().map(crate::body::box_body)
342 | |             }
    | |_____________^ ...so that the type `impl futures_util::Future` will meet its required lifetime bounds
...
349 |   impl_handler!(T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12, T13, T14, T15, T16);
    |   ------------------------------------------------------------------------------------- in this macro invocation
    |
    = note: this error originates in the macro `impl_handler` (in Nightly builds, run with -Z macro-backtrace for more info)
````

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -24,7 +24,7 @@ tower-log = ["tower/log"]
 ws = ["tokio-tungstenite", "sha-1", "base64"]
 
 [dependencies]
-async-trait = "0.1"
+async-trait = "0.1.43"
 bitflags = "1.0"
 bytes = "1.0"
 dyn-clone = "1.0"

